src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/RecordMapper.java

```diff
@@ -0,0 +1,57 @@
+package com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.Record;
+import com.amazonaws.services.dynamodbv2.model.StreamRecord;
+import com.salesforce.dynamodbv2.mt.context.MtAmazonDynamoDbContextProvider;
+import com.salesforce.dynamodbv2.mt.mappers.MtAmazonDynamoDb.MtRecord;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class RecordMapper implements Function<Record, MtRecord> {
```

Comment:
Moved record filtering and mapping logic here, so that this class can be cached as part of `TableMapping`.